### ISSUE TITLE
All temperature and additional echo sensors no longer displaying consistent information

### ISSUE THREAD

**Issue by Reachushere** [alandtse/alexa_media_player#2775]
[home-assistant_alexa_media_2024-12-27T16-53-11.429Z.log](https://github.com/user-attachments/files/18262279/home-assistant_alexa_media_2024-12-27T16-53-11.429Z.log)
**IMPORTANT: Please search the issues, including closed issues, and the [FAQ](https://github.com/alandtse/alexa_media_player/wiki/FAQ) before opening a new issue. The template is mandatory; failure to use it will result in issue closure.**
![Echo Temp](https://github.com/user-attachments/assets/e47e0596-7832-4211-94a9-87f7321c6bd3)

**Describe the bug**

  Until about two months ago, my temperature sensors for my echo devices were able to display on my dashboard without issue.  I have 15 echos and only a handfull of them have temperature sensors.  Then suddenly, they started saying "Unknown" or "Unavailable" but only sometimes.  When I look at the graphs for the sensors, I can see that at random intervals, the temperature is being received by Home Assistant, but for large periods of time, HA is receiving no data.  I have checked the echo devices themselves in the Alexa app, and the temperature sensors are working fine, even when there is no temperature displaying on my dashboard.  So the issue is periodic, specific to Home Assistant, and occurs on multiple Echo devices, and not at the same time.  Sometimes one echo will display the temperature sensor on my dashboard, while another echo won't display it.  It also seems to vary between "Unknown" or "Unavailable" without any reason.  I have reset all of my echo devices, both short and long resets, removed them from Alexa, and it does not result in the temperature displaying on my dashboard again right away - sometimes a few hours later one will show up for an hour or so and then it goes back to Unavailable or Unknown.  The devices are not connected to any bluetooth devices, nor are they part of a speaker set.  This occurs with my Echo Dots and my Echo (Hub).

**To Reproduce**

 Unable to say how to reproduce because it is random.

**Expected behavior**

 Temperatures should display consistently in custom:buttons as they used to
[home-assistant_alexa_media_2024-12-27T16-53-11.429Z.log](https://github.com/user-attachments/files/18262302/home-assistant_alexa_media_2024-12-27T16-53-11.429Z.log)

**Screenshots**

 The red circles show Echo temperature sensors.  The other temperature sensors on the screenshot are not from Echo devices.

**System details**

- Home Assistant version:
Core
2024.12.5
Supervisor
2024.12.0
Operating System
14.1
Frontend
20241127.8

- alexa_media version (from `const.py` or HA startup log): 5.2.0
- alexapy version (from `pip show alexapy` in homeasssistant container or HA startup log): I'm not sure where to find this?

- Is Amazon [2FA/2SV enabled](https://github.com/alandtse/alexa_media_player/wiki/Configuration#enable-two-factor-authentication-for-your-amazon-account)  (y/n): Yes

- Amazon Domain: I don't know what you are asking for here?

**Debug Logs (alexa_media & alexapy)**
Please provide [logs](https://github.com/alandtse/alexa_media_player/wiki/FAQ#how-do-i-enable-debug-logging-for-the-component).

**Additional [context**]()

 My log is 394MB long, it seems to repeat alot of localtuya errors, so I had to cut it down in order to be able to attach it here.

**Comment by ballakers** [comment]
![Screen Shot 2024-12-28 at 08 19 18 AM](https://github.com/user-attachments/assets/e9e96e44-fec1-4c89-9b9e-b688330494bd)
![Screen Shot 2024-12-28 at 08 19 28 AM](https://github.com/user-attachments/assets/71534ef7-19c2-4a94-a031-29f46acf7de4)
It seems the integration forgot the config setup,  my include box was not checked but used to be.  Easy fix for you though, just go to configure and check the box and hit save / may need to reload but should do automagically!

**Comment by Reachushere** [comment]
This is not the issue.  Mine is checked already.  Also, the temperature sensors are not external bluetooth devices.On Dec 28, 2024, at 9:21 AM, Ballakers ***@***.***> wrote:﻿
Screen.Shot.2024-12-28.at.08.19.18.AM.png (view on web)
Screen.Shot.2024-12-28.at.08.19.28.AM.png (view on web)
It seems the integration forgot the config setup,  my include box was not checked but used to be.  Easy fix for you though, just go to configure and check the box and hit save / may need to reload but should do automagically!

**Comment by ballakers** [comment]
Okay then, mine was working previous and was now unchecked and then the temps came right back after I checked the box. Figured same happened to you.

**Comment by Reachushere** [comment]
I wish it were that easy!  But it doesn’t suggest that maybe I should try to reinstall it all again, and recheck that box, so I’ll try that today, thanks :)

**Comment by Reachushere** [comment]
Just tried reinstalling it and rechecking that box and it didn’t help unfortunately.

**Comment by ballakers** [comment]
What Amazon region are you using? And using the latest AMP that was
released this AM or yesterday?

**Comment by Reachushere** [comment]
Canada and yes I had to download the latest when I just reinstalled.

**Comment by ballakers** [comment]
I’m in Canada too, hmm.

**Comment by Reachushere** [comment]
I just keep getting this:

**Comment by Reachushere** [comment]
And this screenshot as well…

Notice the one little blue dot.  So this graph is a bit misleading because I just reinstalled the integration after speaking with you, but this is what it does… it will be black and then I get small blips of blue temperature history, so I know it works sometimes.

**Comment by ballakers** [comment]
Since this is from HACS are you deleting then restarting and then re-adding AMP and then restarting again? You see the two get.html and post.html files in your file editor / VS code?  None of your screenshots came through so maybe try another file type

**Comment by Reachushere** [comment]
Yes, I removed it from HACS and from the main integrations, restart, add in HACS and main integrations under decides and settings, restart.

Attaching two screenshots again.  Did these work?

**Comment by Reachushere** [comment]
![Echo Temp2](https://github.com/user-attachments/assets/c5873a0a-3c64-40c3-9db1-12940383e1ca)
![Echo Temp3](https://github.com/user-attachments/assets/ff5bbd22-8a2d-4dba-8bf5-8c31ec87b729)

**Comment by danielbrunt57** [comment]
> Amazon Domain: I don't know what you are asking for here?

I stripped your debug log down to just the log statements with "alexa":
```
2024-12-26 20:39:04.927 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration alexa_media which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-12-26 20:39:41.335 WARNING (MainThread) [alexapy.alexalogin] Domain amazon.com/ does not match reported account domain amazon.ca; functionality is not likely to work, please fix
2024-12-26 20:39:43.095 WARNING (MainThread) [alexapy.alexalogin] Domain amazon.com/ does not match reported account domain amazon.ca; functionality is not likely to work, please fix
2024-12-26 20:40:06.794 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: {('tuya', '01JACXEE084FSHJPSAWKRY74AB'): 623403.271900383, ('tuya', '01JACXT74HVQCF4Y1J811WGC42'): 623403.272047034, ('alexa_media', '01J4TYQTFBH85T9KV42N9RT64R'): 623403.386597265}
2024-12-26 20:40:11.425 WARNING (MainThread) [alexapy.helpers] alexaapi.get_entity_state((<alexapy.alexalogin.AlexaLogin object at 0x7f1463e12f90>,), {'entity_ids': [']}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()
2024-12-26 21:00:51.335 WARNING (MainThread) [alexapy.helpers] alexaapi.get_entity_state((<alexapy.alexalogin.AlexaLogin object at 0x7f1463e12f90>,), {'entity_ids': ['']}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()
2024-12-26 21:00:51.336 ERROR (MainThread) [custom_components.alexa_media] Error fetching alexa_media data: Error communicating with API: 
2024-12-26 21:21:21.334 WARNING (MainThread) [alexapy.helpers] alexaapi.get_entity_state((<alexapy.alexalogin.AlexaLogin object at 0x7f1463e12f90>,), {'entity_ids': [']}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()
```

You have a region mismatch.  The domain needs to be `amazon.ca` if your login email logs you into that region.

**Comment by Reachushere** [comment]
So I have confirmed I am logged in and have selected Amazon.ca as my region.  After I sent those logs, I reinstalled the integration so I can confirm it is Amazon.ca because I just selected it.

I have posted photos of the issue.  Can you clarify the confusion?  Essentially the temperature sensors for my Echo devices are displaying in the Alexa app but not on my dashboard and I had no issue having them display on my dashboard for the last few years.  Now, they display for maybe 20 minutes a day, at some random time and then they don’t display. 

Let me know if there is anything else I can send.  Thanks :)

Bryn

**Comment by danielbrunt57** [comment]
If you browse to amazon.ca and log in and then select Your Account > Devices, do you see your Amazon devices listed?

![image](https://github.com/user-attachments/assets/e7c5abc9-9584-47d2-9ca3-b4d08714f5fa)

If your devices show as registered in amazon.ca, AMP configuration needs to show amazon.ca as well...

![image](https://github.com/user-attachments/assets/df800678-c652-4537-ac1a-7d811407b6e1)

If the above is correct, scan your debug log for "429" or "Too many requests"...

**Comment by Reachushere** [comment]
So I am getting a great lesson in understanding how logs work, I didn't know many of the things you've already pointed out, so thank you for this.

If I understood you correctly, yes I can see my devices in my Amazon account.  AMP was configured using amazon.ca, I know I did that, and I did it again when I reconfigured after you and I first spoke.  Also my echo's do work with Home Assistant, I can play music through Home Assistant mini media player integration to my echos and they work fine, it is just the temperature sensors and the other data in the red circle in the photos above that doesn't show up on my dashboard lately (or shows up but just for sporadic 10 or 15 minute increments). 

I did find a 429 in the logs.  If those 3 digits after the time are what you need, I also compiled all of the Alexa errors in this handy summary, so I am hoping this might quickly tell you something that doesn't yet make any sense to me but might to you?  Here is what I compiled, a short summary version first, then the detailed logs below the summary:

SHORT SUMMARY VERSION:

429 - Error doing job: Exception in callback _SelectorDatagramTransport

031 - WARNING (MainThread) [alexapy.helpers] alexaapi.get_entity_state((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,)

141 - WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,),{'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()

324 - WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,),{'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()

099 - WARNING (MainThread) [alexapy.helpers] alexaapi.get_entity_state((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,),  Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()

325 - ERROR (MainThread) [homeassistant.config_entries] Error setting up entry (EMAIL ADDRESS) - amazon.ca for media_player ValueError: Config entry (EMAIL ADDRESS) - amazon.ca () for alexa_media.media_player has already been setup!

856 - WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()

965 - ERROR (MainThread) [custom_components.alexa_media] Error fetching alexa_media data: Error communicating with API: 

211 - WARNING (MainThread) [alexapy.helpers] alexaapi.get_entity_state((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>, Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()

212 - ERROR (MainThread) [custom_components.alexa_media] Error fetching alexa_media data: Error communicating with API: 

818 - WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()

024 - WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()

990 - WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()


------------------------------------------------
MORE DETAILED ERROR INFO FOR EACH OF THE ABOVE:


2024-12-30 22:08:29.429 ERROR (MainThread) [homeassistant] Error doing job: Exception in callback _SelectorDatagramTransport._read_ready() (None)
Traceback (most recent call last):
  File "/config/custom_components/localtuya/discovery.py", line 65, in datagram_received
    data = decrypt_udp(data)
  File "/config/custom_components/localtuya/discovery.py", line 30, in decrypt_udp
    return _unpad(decryptor.update(message) + decryptor.finalize()).decode()
                                              ~~~~~~~~~~~~~~~~~~^^
ValueError: The length of the provided data is not a multiple of the block length.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.13/asyncio/events.py", line 89, in _run
    self._context.run(self._callback, *self._args)
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/selector_events.py", line 1240, in _read_ready
    self._protocol.datagram_received(data, addr)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
  File "/config/custom_components/localtuya/discovery.py", line 67, in datagram_received
    data = data.decode()
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 2: invalid start byte





2024-12-30 21:59:07.429 WARNING (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id eb57012df1d06cfdfbwgtt




2024-12-31 05:56:50.031 WARNING (MainThread) [alexapy.helpers] alexaapi.get_entity_state((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'entity_ids': []}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()




2024-12-31 06:28:34.141 WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()




2024-12-31 08:56:18.324 WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()




2024-12-30 21:59:11.099 WARNING (MainThread) [alexapy.helpers] alexaapi.get_entity_state((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'entity_ids': ]}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()




2024-12-30 21:59:24.325 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry (EMAIL ADDRESS) - amazon.ca for media_player
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 640, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/media_player/__init__.py", line 456, in async_setup_entry
    return await hass.data[DATA_COMPONENT].async_setup_entry(entry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 186, in async_setup_entry
    raise ValueError(
    ...<2 lines>...
    )
ValueError: Config entry (EMAIL ADDRESS) - amazon.ca () for alexa_media.media_player has already been setup!




2024-12-31 00:26:37.856 WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()





2024-12-31 00:37:07.965 ERROR (MainThread) [custom_components.alexa_media] Error fetching alexa_media data: Error communicating with API: 





2024-12-31 02:25:56.211 WARNING (MainThread) [alexapy.helpers] alexaapi.get_entity_state((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'entity_ids': []}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()




2024-12-31 02:25:56.212 ERROR (MainThread) [custom_components.alexa_media] Error fetching alexa_media data: Error communicating with API: 




2024-12-31 02:36:39.818 WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()




2024-12-31 03:08:35.024 WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()




2024-12-31 05:56:50.031 WARNING (MainThread) [alexapy.helpers] alexaapi.get_entity_state((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'entity_ids': []}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()



2024-12-31 06:28:34.141 WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()



2024-12-31 06:39:17.990 WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()





2024-12-31 08:56:18.324 WARNING (MainThread) [alexapy.helpers] alexaapi.get_customer_history_records((<alexapy.alexalogin.AlexaLogin object at 0x7f3b9ff38050>,), {'max_record_size': 10}): Timeout error occurred accessing AlexaAPI: An exception of type CancelledError occurred. Arguments:
()

**Comment by Reachushere** [comment]
Each of the above are specific to Alexa errors

**Comment by Reachushere** [comment]
![Echo Temp4](https://github.com/user-attachments/assets/6afaad80-e47b-4c7e-ab0f-c4487380720b)

**Comment by Reachushere** [comment]
![Echo Temp5](https://github.com/user-attachments/assets/fadd1e98-5ef6-44b1-aad4-8bb71a82ecab)

**Comment by Reachushere** [comment]
Interestingly, a moment ago where the temperature is circled in red here, it was actually showing me a temperature, and it was also displayed on my dashboard.  The moment I clicked on it to pull up the graph in the second picture, it went unavailable again.  But you can see from the graph how sporadic it is working anyway...
![Echo temp7](https://github.com/user-attachments/assets/bda83a29-474f-4736-a915-0773cceec51c)
![Echo Temp6](https://github.com/user-attachments/assets/7f077f73-022a-4d02-9c7a-af702c675ba2)

**Comment by Reachushere** [comment]
> "You have a region mismatch. The domain needs to be amazon.ca if your login email logs you into that region and is where the devices are registered."`

2024-12-26 20:39:43.095 WARNING (MainThread) [alexapy.alexalogin] Domain amazon.com/ does not match reported account domain amazon.ca; functionality is not likely to work, please fix

I don't understand why I am seeing amazon.com in the logs in those examples you gave, yet you can see from the screenshot above that I used amazon.ca as the domain.   How do I fix this?  I've reinstalled the integration and confirmed I chose amazon.ca, and I log in using amazon.ca as you can see from the screenshot above.

**Comment by andywarburton** [comment]
I have also noticed this behavior in my system recently - I used to get consistent temperature readings as you can see in the graph and then suddenly, it just stopped. My account is in the UK. If I can do anything to help debug, please let me know.

![Screenshot 2025-01-04 013556](https://github.com/user-attachments/assets/3f3f0d1e-775e-40c4-a2f2-d5309032d2cb)

**Comment by andywarburton** [comment]
![IMG_5240](https://github.com/user-attachments/assets/cbbc0066-3887-4423-a92b-51e84a0a0096)

Strangely, the sensors came back online for a brief amount of time today but I haven't had any readings for the past three hours. Do you think we could be hitting some API limits? (too many requests?). I'll run off some logs to see if it helps at all.

**Comment by andywarburton** [comment]
[home-assistant_alexa_media_2025-01-04T16-35-07.224Z.log](https://github.com/user-attachments/files/18307546/home-assistant_alexa_media_2025-01-04T16-35-07.224Z.log)

Log file attatched - sorry, I have no idea how to read this or if it's any use!

**Comment by Reachushere** [comment]
I think if they were the case, then the times they come online would be longer, no?  I can’t imagine their limit lasts 20 minutes in 4 days, which is all that mine have been on for oftentimes lately.On Jan 4, 2025, at 11:24 AM, Andy Warburton ***@***.***> wrote:﻿
IMG_5240.JPG (view on web)
Strangely, the sensors came back online for a brief amount of time today but I haven't had any readings for the past three hours. Do you think we could be hitting some API limits? (too many requests?). I'll run off some logs to see if it helps at all.

**Comment by Reachushere** [comment]
> [home-assistant_alexa_media_2025-01-04T16-35-07.224Z.log](https://github.com/user-attachments/files/18307546/home-assistant_alexa_media_2025-01-04T16-35-07.224Z.log)
> 
> 
> 
> Log file attatched - sorry, I have no idea how to read this or if it's any use!

I only know how to read them with regards to what the other person asked. I did a search on the word Alexa, and then for each line that had an issue, right after the date there are three numbers.  He said to look for the ones that said 429.

Haven't heard anything more from him on what that means though, I provided mine below.

**Comment by andywarburton** [comment]
![image](https://github.com/user-attachments/assets/b2ae7eb4-7e08-4ffc-b665-9f590f65c693)

Weirdly, my sensors have just come back online... almost exactly the same time they did yesterday!

**Comment by Reachushere** [comment]
> ![image](https://github.com/user-attachments/assets/b2ae7eb4-7e08-4ffc-b665-9f590f65c693)
> 
> 
> 
> Weirdly, my sensors have just come back online... almost exactly the same time they did yesterday!

Here is my last 48 hours.. those tiny blue dots is all!  Are yours staying online?


[image](https://github.com/user-attachments/assets/d893aa35-619e-4990-a011-f049d0a71251)

**Comment by Reachushere** [comment]
> ![image](https://github.com/user-attachments/assets/b2ae7eb4-7e08-4ffc-b665-9f590f65c693)
> 
> 
> 
> Weirdly, my sensors have just come back online... almost exactly the same time they did yesterday!

Actually here is a comparative one, and they all come online at the same time and go offline at the same time.

I wonder how this compares with our time zones, I am in Toronto so Eastern time.  Though yours go online for much longer than mine so I can't imagine it is a global issue.!

[image](https://github.com/user-attachments/assets/72e12f30-4271-4df3-828a-2f57328667df)

**Comment by andywarburton** [comment]
![IMG_5262](https://github.com/user-attachments/assets/cdb4ae7b-20d8-43ec-8a4c-5e39a07768eb)

Strangely, today so far I have had a solid run of readings  since around 11pm!

**Comment by Reachushere** [comment]
> ![IMG_5262](https://github.com/user-attachments/assets/cdb4ae7b-20d8-43ec-8a4c-5e39a07768eb)
> 
> 
> 
> Strangely, today so far I have had a solid run of readings  since around 11pm! 

So that's interesting. I am the same as you on my Echo, but my two echo dots are not getting readings.  Have you done anything different?

![image](https://github.com/user-attachments/assets/1f54fec8-861d-46b6-a698-f84a6cf6020d)

**Comment by danielbrunt57** [comment]
Check your logs for `returned 429:Too Many Requests`

**Comment by Reachushere** [comment]
> Check your debug logs for `returned 429:Too Many Requests`

I posted my log above with regards to the 429 number.  Not sure what that means though.  There were a few examples of 429 but they don't say "too many requests" and also I don't know what to do about them?

**Comment by danielbrunt57** [comment]
> for each line that had an issue, right after the date there are three numbers. He said to look for the ones that said 429.

The "429" you pointed out is the number of milliseconds in the date-timestamp for the log entry:
`2024-12-30 22:08:29.429` => `yyyy-mm-dd hh:mm:ss.ms`

You need to look for any `alexa` log entries where the HTTP returned response is the error code `429` which means `Too Many Requests`.  Just look for `429:Too Many Requests`.  If you are not seeing any then I don't know what the issue might be...

**Comment by Reachushere** [comment]
> > for each line that had an issue, right after the date there are three numbers. He said to look for the ones that said 429.
> 
> 
> 
> The "429" you pointed out is the number of milliseconds in the date-timestamp for the log entry:
> 
> `2024-12-30 22:08:29.429` => `yyyy-mm-dd hh:mm:ss.ms`
> 
> 
> 
> You need to look for any `alexa` log entries where the HTTP returned response is the error code `429` which means `Too Many Requests`.  Just look for `429:Too Many Requests`.  If you are not seeing any then I don't know what the issue might be...

Oh ok, I don't have any other 429's in my log so I guess that can't be it.

**Comment by danielbrunt57** [comment]
The normal sequence in the scheduled polling is as follows:
First, the echo devices and speaker groups:
```
2025-01-06 01:21:18.486 DEBUG (MainThread) [custom_components.alexa_media] Echo Show 8: Locale en-ca timezone America/Vancouver
2025-01-06 01:21:18.486 DEBUG (MainThread) [custom_components.alexa_media] Echo Show 8: DND False
2025-01-06 01:21:18.486 DEBUG (MainThread) [custom_components.alexa_media.media_player] d****l@b******a: Refreshing Echo Show 8
2025-01-06 01:21:18.487 DEBUG (MainThread) [custom_components.alexa_media.media_player] d****l@b******a: Echo Show 8: Last_called check: self: G************326 reported: 

2025-01-06 01:21:18.488 DEBUG (MainThread) [custom_components.alexa_media] Norton Court: DND False
2025-01-06 01:21:18.489 DEBUG (MainThread) [custom_components.alexa_media.media_player] d****l@b******a: Refreshing Norton Court
2025-01-06 01:21:18.489 DEBUG (MainThread) [custom_components.alexa_media.media_player] d****l@b******a: Norton Court: Last_called check: self: 9****************************341 reported: 

2025-01-06 01:21:18.490 DEBUG (MainThread) [custom_components.alexa_media] Everywhere: DND False
2025-01-06 01:21:18.490 DEBUG (MainThread) [custom_components.alexa_media.media_player] d****l@b******a: Refreshing Everywhere
2025-01-06 01:21:18.491 DEBUG (MainThread) [custom_components.alexa_media.media_player] d****l@b******a: Everywhere: Last_called check: self: 5****************************da7 reported: 

2025-01-06 01:21:18.491 DEBUG (MainThread) [custom_components.alexa_media] This Device: Locale en-us timezone None
2025-01-06 01:21:18.492 DEBUG (MainThread) [custom_components.alexa_media] This Device: DND False
2025-01-06 01:21:18.492 DEBUG (MainThread) [custom_components.alexa_media.media_player] d****l@b******a: Refreshing This Device
2025-01-06 01:21:18.492 DEBUG (MainThread) [custom_components.alexa_media.media_player] d****l@b******a: This Device: Last_called check: self: a****************************f2f reported: 
```
Summary:
```
2025-01-06 01:21:18.493 DEBUG (MainThread) [custom_components.alexa_media] d****l@b******a: Existing: [<entity media_player.office_echo_dot_left=standby>, <entity media_player.office_echo_dot_right=standby>, <entity media_player.kitchen_echo_dot=standby>, <entity media_player.living_room_echo_dot_right=standby>, <entity media_player.living_room_echo_dot_left=standby>, <entity media_player.garage_echo_dot=standby>, <entity media_player.bedroom_echo_dot_right=standby>, <entity media_player.bedroom_echo_dot_left=standby>, <entity media_player.echo=standby>, <entity media_player.bedroom=standby>, <entity media_player.living_room=standby>, <entity media_player.office=standby>, <entity media_player.echo_show_8=standby>, <entity media_player.norton_court=standby>, <entity media_player.everywhere=standby>, <entity media_player.this_device=standby>] New: []; Filtered out by mode (): []
2025-01-06 01:21:18.503 DEBUG (MainThread) [custom_components.alexa_media] Finished fetching alexa_media data in 6.064 seconds (success: True)
```
Then my two temperature entities:
1. Echo with temp sensor:
```
2025-01-06 01:21:18.504 DEBUG (MainThread) [custom_components.alexa_media.alexa_entity] parse_temperature_from_coordinator: {'value': 19.5, 'scale': 'CELSIUS'}
2025-01-06 01:21:18.504 DEBUG (MainThread) [custom_components.alexa_media.sensor] TemperatureSensor value: 19.5
2025-01-06 01:21:18.504 DEBUG (MainThread) [custom_components.alexa_media.sensor] TemperatureSensor scale: CELSIUS
2025-01-06 01:21:18.504 DEBUG (MainThread) [custom_components.alexa_media.sensor] Coordinator update: Living Room Air Temperature: 19.5 °C
```
2. AQM Temp sensor:
```
2025-01-06 01:21:18.506 DEBUG (MainThread) [custom_components.alexa_media.alexa_entity] parse_temperature_from_coordinator: {'value': 19.3, 'scale': 'CELSIUS'}
2025-01-06 01:21:18.506 DEBUG (MainThread) [custom_components.alexa_media.sensor] TemperatureSensor value: 19.3
2025-01-06 01:21:18.506 DEBUG (MainThread) [custom_components.alexa_media.sensor] TemperatureSensor scale: CELSIUS
2025-01-06 01:21:18.506 DEBUG (MainThread) [custom_components.alexa_media.sensor] Coordinator update: Echo Temperature: 19.3 °C
```
Then the sensors for the 16 echo devices:
```
2025-01-06 01:21:36.242 DEBUG (MainThread) [custom_components.alexa_media.sensor] Sensor value ('142f700d-c9d6-39dd-bf92-dbdf5d88b3c9', {'alarmLabel': None, 'alarmTime': 0, 'assigner': None, 'automationId': None, 'createdDate': 1735478634717, 'deferredAtTime': None, 'deliveryMetadata': None, 'deliveryType': None, 'deviceName': None, 'deviceSerialNumber': 'G091ET10051402M6', 'deviceType': 'A2H4LV5GIZ1JFT', 'extensibleAttribute': None, 'followUpMetadata': None, 'geoLocationTriggerData': None, 'id': 'A2H4LV5GIZ1JFT-G091ET10051402M6-142f700d-c9d6-39dd-bf92-dbdf5d88b3c9', 'instanceState': None, 'lastOccurrenceTimeInMilli': 0, 'lastTriggerTimeInUtc': None, 'lastUpdatedDate': 1735493962404, 'loopCount': None, 'mrmGroupContext': None, 'musicAlarmId': None, 'musicEntity': None, 'notificationIndex': '142f700d-c9d6-39dd-bf92-dbdf5d88b3c9', 'originalDate': '2024-12-29', 'originalDurationInMillis': 0, 'originalTime': '09:30:00.000', 'personProfile': None, 'primaryDeviceState': None, 'provider': None, 'rRuleData': None, 'recurrenceEligibility': False, 'recurringPattern': None, 'remainingTime': 0, 'reminderAppLabel': None, 'reminderLabel': None, 'reminderSubLabel': None, 'showDeregisteredWarning': None, 'skillInfo': None, 'sleepTrackingAlarmProperties': {'sleepTrackingEnabled': False}, 'snoozedToTime': None, 'sound': {'displayName': 'Clarity', 'folder': None, 'id': 'system_alerts_melodic_05', 'isChildDirected': False, 'numAssets': 0, 'pricing': None, 'providerId': 'ECHO', 'sampleUrl': 'https://s3.amazonaws.com/deeappservice.prod.notificationtones/system_alerts_melodic_05.mp3', 'sampleVideoUrl': None, 'skillUrl': None}, 'status': 'OFF', 'targetPersonProfiles': None, 'taskMetadata': None, 'timeZoneId': None, 'timerLabel': None, 'tmwNotificationProperties': None, 'triggerTime': 0, 'type': 'Alarm', 'version': '5', 'date_time': datetime.datetime(2024, 12, 29, 9, 30, tzinfo=zoneinfo.ZoneInfo(key='America/Vancouver'))})
```

**Comment by Swordnut** [comment]
Hi, just in case you haven't tried it, try removing a device from your amazon account, making a new account and adding it to that. then set up your new account in the integration. There are issues in the amazon backend that interrupt or block entity accessibility

**Comment by jojo01fr** [comment]
This "Include devices connected via Echo" gets unchecked on each AMP version update, perhaps it is time to fix some stuff ? :)

**Comment by jojo01fr** [comment]
I disagree with your disagreement but we both can lead to an agreement if we could expose this setting as an entity on HA as well as AMP version to be able to survey and make a final statement. Would someone have an idea how to do it ? I dont.

**Comment by danielbrunt57** [comment]
The Alexa Media Player integration config entry data is stored in `/config/.storage/core.config` entries.  
The config entry (or entries if you've configured more than one account) in that file contains `"domain":"alexa_media"`.
The field name in question for the config entry is `"extended_entity_discovery"` which will be either `:"true"` or `:"false`.

I just resurrected my HA system after a month in storage while I moved.
I had 26 updates to do, one of which was AMP.
The value for "extended_entity_discovery" is still set to "true" in the HA config file and...

<img width="376" alt="Image" src="https://github.com/user-attachments/assets/4258c3fd-7063-41d5-932d-5c4ccbdaa46d" />

**Comment by danielbrunt57** [comment]
> if we could expose this setting as an entity on HA as well as AMP version to be able to survey and make a final statement. Would someone have an idea how to do it ?

Why would anyone want to waste their their time coding such a thing?
All you have to do is go to `Integrations` and select `Alexa Media Player` to see the current AMP version number.
Then select `CONFIGURE` to see the setting for `Include devices connected to echo`.

**Comment by jojo01fr** [comment]
To detect WHEN it goes off so I can explain WHY it goes off.

**Comment by danielbrunt57** [comment]
I think you've gone down the wrong rabbit hole in your evaluation of the issue & cause and you've provided zero evidence of your conclusion
That setting is not going to change unless you remove the AMP config entry & re-add it with a different setting.
Even if you were to manually edit `core.config_entries`, the flag in AMP's memory won't change until AMP is reloaded or HA is restarted.  `core.config_entries` doesn't get touched when upgrading AMP. Upgrading AMP simply replaces the files in `configuration/alexa_media` and HA has to be reloaded to instantiate the new files.  There's no code in HA or AMP that would touch AMP except for an integration reconfiguration.

If I were you and suspected updating is somehow changing it, I would first ensure that the flag in the UI view of the config entry and also in `core_config_entries` is currently "True", then update AMP and recheck the setting, then restart HA and check the setting once again.  If I'm somehow wrong, I'll happily eat humble pie and continue down this rabbit hole with you.

**Comment by jojo01fr** [comment]
What I can say is that I am not the only one person who has the issue. I do not believe in Santa Claus or in ghosts but something is removing that setting on my AMP from time to time and since I remember I already had to re enable it 3 times in the last 6 months. No HA backup restore, no nothing. I dont have nabucasa, I've integrated AMP through Amazon developper like explained and a part from this issue it works great, thanks to the coders.

**Comment by Swordnut** [comment]
Hi,

I think I get it. You want to be able to see if the temp state dropouts are correlated to any apparent changes to the config? But standard sensors can't pick up config entries.

You should* be able to use a python script to automate a check on the config setting by reading the file and returning the value of the option. Create a sensor for the value returned from that script to record results over time and that should give you the info you are after.

Here is a starter from GPT

Step 1: Create the Python Script

Create a file called check_alexa_discovery.py in your config/python_scripts

 ```
import json

config_path = "/config/.storage/core.config_entries"

try:
    with open(config_path, "r") as f:
        data = json.load(f)
        for entry in data.get("data", {}).get("entries", []):
            if entry.get("domain") == "alexa_media":
                options = entry.get("options", {})
                value = options.get("extended_entity_discovery", False)
                print(str(value).lower())
                break
        else:
            print("false")
except Exception as e:
    print("false")
```

Step 2: Set Up the Sensor

Add this to your configuration.yaml:

```
sensor:
  - platform: command_line
    name: Alexa Extended Discovery State
    command: "python3 /config/python_scripts/check_alexa_discovery.py"
    scan_interval: 300 #5mins or your preference 
    value_template: "{{ value | lower == 'true' }}"
```

Step 3: Recorder Settings for Long-Term History

Make sure the sensor is recorded, again in your configuration.yaml:

```
recorder:
  include:
    entities:
      - sensor.alexa_extended_discovery_state
  purge_keep_days: 90 # or your preferred time period
```

*a word often used to hedge against future disappointment 😀
